torchao/experimental/kernels/cpu/aarch64/kleidi/kai_matmul_clamp_f32_qai8dxp_qsi4c32p.h

```diff
@@ -148,5 +159,99 @@ void prepare_weight_data(
       /*qparams=*/&qparams);
 }
 
+
+size_t get_preferred_alignement() {
+  return 16;
+}
+
+
+#define DEFINE_WEIGHT_DATA_FNS(nr, kr, sr)                                          \
+    size_t weight_data_size_nr##nr##_kr##kr##_sr##sr(int n, int k, int group_size) {  \
+        return weight_data_size(nr, kr, sr, n, k, group_size);          \
+    }                                                                               \
+    void prepare_weight_data_nr##nr##_kr##kr##_sr##sr(                                \
+        void* weight_data,                                                          \
+        int n,                                                                      \
+        int k,                                                                      \
+        int group_size,                             \
+        const int8_t* weight_qvals, \
+        const float* weight_scales, \
+        const int8_t* weight_zeros, \
+        const float* bias) { \
+        prepare_weight_data(nr, kr, sr, weight_data, n, k, group_size, weight_qvals, weight_scales, weight_zeros, bias); \
+    }
+
+#define DEFINE_ACTIVATION_DATA_FNS(mr, kr, sr) \
+    size_t activation_data_size_mr##mr##_kr##kr##_sr##sr(int m, int k, int group_size) {  \
+        (void)group_size; \
+        return activation_data_size(mr, kr, sr, m, k);        \
+    } \
+    void prepare_activation_data_mr##mr##_kr##kr##_sr##sr(void* activation_data, int m, int k, int group_size, const float* activations) { \
+        (void)group_size; \
+        prepare_activation_data(mr, kr, sr, activation_data, m, k, activations); \
+    }
+
+// TODO: first and suffix need to be better, e.g., parametrized by mr, nr, etc
+// But I don't quite follow the naming convention for KleidiAI
+#define DEFINE_KERNEL_FNS(first, suffix) \
+  namespace impl_##suffix { \
+  const Ukernel get_ukernel() { \
+  return Ukernel{ \
+  .get_m_step = kai_get_m_step_matmul_clamp_f32_qai8dxp##first##_qsi4c32p##suffix, \
```

Comment:
Passing in first ir lhs as `qai8dxp1x8` instead of `1x8` is better for (1) meaningful, (2) can cover channel wise 4b quant i.e. QC4W as well.

Also `suffix` should be three different things, `rhs` + `output tile x kacc` + `isa`, where rhs is not `8x8` but `qsi4c32p4x8`.

```suggestion
#define DEFINE_KLEIDI_KERNEL_FN(lhs, suffix) \
  namespace impl_##suffix { \
  const Ukernel get_ukernel() { \
  return Ukernel{ \
  .get_m_step = kai_get_m_step_matmul_clamp_f32_##lhs##_##suffix, \
```